src/dmd/dsymbolsem.d

```diff
@@ -3735,7 +3735,18 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         if (dd.ident == Id.dtor && dd.semanticRun < PASS.semantic)
             ad.dtors.push(dd);
         if (!dd.type)
-            dd.type = new TypeFunction(null, Type.tvoid, false, LINK.d, dd.storage_class);
+        {
+            Parameters* params = null;
+            ClassDeclaration cd = ad.isClassDeclaration();
+            if (cd && cd.classKind == ClassKind.cpp && !Target.cppDeletingDestructor)
```

Comment:
How can I also detect if `ad` is an `extern(C++) struct`??

Reply:
Perhaps add a `linkage` instance variable to `StructDeclaration` just as it exists for `Declaration` and `classKind` exists for `ClassDeclaration`.